packages/components/src/components/lineage/vis/VisGraphGenerator.ts

```diff
@@ -135,32 +135,34 @@ export interface VisGraphClusterNode {
 }
 
 export function isBasicNode(item: VisGraphNodeType): item is VisGraphNode {
-    return item.kind === 'node';
+    return item && item.kind === 'node';
 }
 
 export function isCombinedNode(item: VisGraphNodeType): item is VisGraphCombinedNode {
-    return item.kind === 'combined';
+    return item && item.kind === 'combined';
 }
 
 export function isClusterNode(item: VisGraphNodeType): item is VisGraphClusterNode {
-    return item.kind === 'cluster';
+    return item && item.kind === 'cluster';
 }
 
-export class VisGraphOptions extends Record({
-    edges: undefined,
-    nodes: undefined,
-    options: {},
-    makeClusters: undefined,
-    initialSelection: undefined,
-}) {
-    edges?: DataSet<Edge>;
-    nodes?: DataSet<VisGraphNode | VisGraphCombinedNode>;
-    options?: { [key: string]: any };
-    makeClusters?: (network: Network, options?: VisGraphOptions) => void;
-    initialSelection?: string[];
-
-    constructor(values?: { [key: string]: any }) {
-        super(values);
+export interface IVisGraphOptions {
+    edges: DataSet<Edge>;
+    initialSelection: string[];
+    nodes: DataSet<VisGraphNode | VisGraphCombinedNode>;
+    options: { [key: string]: any };
+}
+
+export class VisGraphOptions implements IVisGraphOptions {
+    [immerable] = true;
+
+    readonly edges: DataSet<Edge>;
+    readonly initialSelection: string[];
+    readonly nodes: DataSet<VisGraphNode | VisGraphCombinedNode>;
+    readonly options: { [key: string]: any };
+
+    constructor(config?: Partial<IVisGraphOptions>) {
+        Object.assign(this, { ...config });
```

Comment:
I think this can just be `Object.assign(this, config);`

Reply:
Fixed.